File: Backend/Challenge.Domain/Entities/Pokedex.cs
using Challenge.Domain.Aggregates;
using Challenge.Domain.Interfaces.Aggregates;
using Challenge.Domain.Interfaces.Entities;

namespace Challenge.Domain.Entities
{
    public class Pokedex : IPokedex
    {
        public int Count { get; set; }

        public string? Next { get; set; }

        public string? Previous { get; set; }

        public List<PokeShortDescription> Results { get; set; } = new List<PokeShortDescription>();

        public Pokedex(int count, string next, string previous, List<PokeShortDescription> results)
        {
            Count = count;
            Next = next; 
            Previous = previous;
            Results = results;
        }
        public Pokedex() { }
    }
}

File: JsonFiles/api/models/Pokedex.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { PokeShortDescription } from './PokeShortDescription';

export type Pokedex = {
    count?: number;
    next?: string | null;
    previous?: string | null;
    results?: Array<PokeShortDescription> | null;
};
